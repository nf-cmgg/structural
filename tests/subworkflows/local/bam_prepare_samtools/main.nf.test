nextflow_workflow {

    name "Test Workflow BAM_PREPARE_SAMTOOLS"
    script "subworkflows/local/bam_prepare_samtools/main.nf"
    workflow "BAM_PREPARE_SAMTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_prepare_samtools"

    test("homo sapiens - [cram, []], fasta, fai") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    []
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.crams,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match()}
            )
        }

    }

    test("homo sapiens - [[cram, crai], [cram, []]], fasta, fai - One sample") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ], [
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram2, checkIfExists: true),
                    []
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    file(workflow.out.crams[0][1]).name,
                    file(workflow.out.crams[0][2]).name,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match()}
            )
        }

    }

    test("homo sapiens - [[cram, crai], [cram, []]], fasta, fai - Two samples") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ], [
                    [id:"test2", sample:"test2", sex:"male"],
                    file(params.cram2, checkIfExists: true),
                    []
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.crams,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match()}
            )
        }

    }

}
