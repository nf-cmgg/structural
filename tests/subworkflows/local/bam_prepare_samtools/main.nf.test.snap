{
    "homo sapiens - [cram, []], fasta, fai": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male"
                    },
                    "/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/cram/test.paired_end.sorted.cram",
                    "test.paired_end.sorted.cram.crai:md5,31ffc1b11f53210e4ac0d55092de0652"
                ]
            ],
            1
        ],
        "timestamp": "2023-11-28T13:19:21.600361731"
    },
    "homo sapiens - [[cram, crai], [cram, []]], fasta, fai - One sample": {
        "content": [
            "test.cram",
            "test.cram.crai",
            2
        ],
        "timestamp": "2023-11-28T13:24:23.217939263"
    },
    "homo sapiens - [[cram, crai], [cram, []]], fasta, fai - Two samples": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "sample": "test",
                        "sex": "male"
                    },
                    "/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/cram/test.paired_end.sorted.cram",
                    "/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/cram/test.paired_end.sorted.cram.crai"
                ],
                [
                    {
                        "id": "test2",
                        "sample": "test2",
                        "sex": "male"
                    },
                    "/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/cram/test2.paired_end.sorted.cram",
                    "test2.paired_end.sorted.cram.crai:md5,633a024a94932189bd5fbc84f62ce6ac"
                ]
            ],
            1
        ],
        "timestamp": "2023-11-28T13:22:41.340593018"
    }
}